pallets/creditcoin/src/tests.rs

```diff
@@ -339,8 +352,39 @@ fn register_address_pre_existing() {
 }
 
 #[test]
-/// Address too long.
-fn register_address_malformed_address() {
+fn register_address_should_error_when_not_signed() {
+	ExtBuilder::default().build_and_execute(|| {
+		let (_who, address, ownership_proof, _) = generate_address_with_proof("owner");
+		let blockchain = Blockchain::Rinkeby;
+
+		assert_noop!(
+			Creditcoin::register_address(Origin::none(), blockchain, address, ownership_proof),
+			BadOrigin,
+		);
+	})
+}
+
+#[test]
+fn register_address_should_error_when_using_wrong_ownership_proof() {
+	ExtBuilder::default().build_and_execute(|| {
+		let (who, address, _ownership_proof, _) = generate_address_with_proof("owner");
+		let (_who2, _address2, ownership_proof2, _) = generate_address_with_proof("bogus");
+
+		let blockchain = Blockchain::Rinkeby;
+		assert_noop!(
+			Creditcoin::register_address(
+				Origin::signed(who.clone()),
+				blockchain,
+				address,
+				ownership_proof2
+			),
+			crate::Error::<Test>::OwnershipNotSatisfied
+		);
+	})
+}
+
+#[test]
+fn register_address_should_error_when_address_too_long() {
```

Comment:
@CAGS295 I need some help reproducing 2 of the error conditions:

- `InvalidSignature` - https://github.com/gluwa/creditcoin/blob/dev/pallets/creditcoin/src/lib.rs#L650
- `MalformedExternalAddress` - https://github.com/gluwa/creditcoin/blob/dev/pallets/creditcoin/src/lib.rs#L667

What input data do I need to pass in order to trigger those conditions ?

Reply:
covered on slack, ping me again if you need more info.